docs/src/design/proposals/APIRedesign_PeteRichards.md

```diff
@@ -0,0 +1,251 @@
+
+Clearly specify supported extension points (e.g. persistence, model providers, telemetry providers, routes, time systems), but don't claim that the system has a clear and perfect repeatable solution for unknown extension types.  New extension categories can be implemented in whatever way makes sense, without prescribing "the one and only system for managing extensions".
+
+The underlying problem here is we are predicting needs for extension points where none exist-- if we try and design the extension system before we know how it is used, we design the wrong thing and have to rewrite it later.
+
+## Composite services should not be the default
+
+Understanding composite services, and describing services as composite services can confuse developers.  Aggregators are implemented once and forgotten, while decorators tend to be hacky, brittle solutions that are generally needed to avoid circular imports.  While composite services are a useful construct, it reduces interface depth to implement them as registries + typed providers.
+
+You can write a provider (provides "thing x" for "inputs y") with a simple interface.  A provider has two or more methods:
+* a method which takes "inputs y" and returns True if it knows how to provide "thing x", false otherwise.
+* one or more methods which provide "thing x" for objects of "inputs y".  
+
+Actually checking whether a provider can respond to a request before asking it to do work allows for faster failure and clearer errors when no providers match the request.
+
+## Get rid of views, representations, and templates.
+
+Templates are an implementation detail that should be handled by module loaders.  Views and representations become "components," and a new concept, "routes", is used to exposing specific views to end users.
+
+`components` - building blocks for views, have clear inputs and outputs, and can be coupled to other components when it makes sense.  (e.g. parent-child components such as menu and menu item), but should have ZERO knowledge of our data models or telemetry apis.  They should define data models that enable them to do their job well while still being easy to test.
+
+`routes` - a view type for a given domain object, e.g. a plot, table, display layout, etc.  Can be described as "whatever shows in the main screen when you are viewing an object."  Handle loading of data from a domain object and passing that data to the view components.  Routes should support editing as it makes sense in their own context.
+
+To facilitate testing:
+
+* routes should be testable without having to test the actual view.
+* components should be independently testable with zero knowledge of our data models or telemetry APIs.
+
+Component code should be organized side by side, such as:
+
+```
+app
+|- components
+  |- productDetail
+  | |- productDetail.js
+  | |- productDetail.css
+  | |- productDetail.html
+  | |- productDetailSpec.js
+  |- productList
+  |- checkout
+  |- wishlist
+```
```

Comment:
Sold. Nicely replaces the role of the bundle as a mechanism for colocating related scripts and assets.

This puts a bit more responsibility on our build, but that is not relevant from an API perspective, and we're more than ready to put more responsibility into the build.